utils/src/main/scala/za/co/absa/enceladus/utils/transformations/DeepArrayTransformations.scala

```diff
@@ -652,11 +789,84 @@ object DeepArrayTransformations {
     */
   private def isLeafElement(path: Seq[String]): Boolean = path.lengthCompare(2) < 0
+
+  /**
+    * Converts from a simple transformation to en extended transformation.
+    * In the extended expression a function for getting columns by fully qualified names is ignored.
+    *
+    * @param transformation A transformation function that takes a column and returns a column.
+    * @return An extended transformation that takes a column and a function hat returns columns from fully qualified
+    *         names and returns a column.
+    */
+  private def toExtendedTransformation(transformation: TransformFunction): ExtendedTransformFunction = {
+    (field, _) => transformation(field)
+  }
+
+  /**
+    * The method takes a field and a parent field, both fully qualified. The method does the following:
+    * <ul>
+    *   <li>If the input field is a child of the parent field the path is split into parent and child parts and these
+    *   parts are returned as a pair.</li>
+    *   <li>If the input field is not a child of the parent field the method returns a pair of empty string and
+    *   the full qualified input field name.</li>
+    * </ul>
+    *
+    * @param field       A fully qualified input field name
+    * @param parentField A parent field to split in case the input field is its child
+    * @return A pair consisting of the parent and the child field parts
+    */
+  private[transformations] def splitParentField(field: String, parentField: String): (String, String) = {
+    val fixedField = field.trim
+    val fixedParentField = parentField.trim
+
+    if (fixedField == fixedParentField) {
+      (fixedField, "")
+    } else {
+      val parentFieldWithDot = if (fixedParentField.endsWith(".")) fixedParentField else fixedParentField + "."
+      val parentFieldWithoutDot = if (fixedParentField.endsWith(".")) fixedParentField.dropRight(1) else fixedParentField
+
+      if (fixedField.startsWith(parentFieldWithDot)) {
+        (parentFieldWithoutDot, fixedField.substring(parentFieldWithDot.length))
+      } else {
+        ("", fixedField)
+      }
+    }
+  }
+
+  /**
+    * The method takes a field and a list of parent fields, all fully qualified. The method does the following:
+    * <ul>
+    *  <li>If the input field is a child of any of the parent fields, a deepest parent is selected, and the path is
+    *  split into parent and child parts and these parts are returned as a pair.</li>
+    *  <li>If the input field is not a child of any of the listed parents the method returns a pair of empty string and
+    *  the full qualified input field name.</li>
+    * </ul>
+    *
+    * @param field        A fully qualified input field name
+    * @param parentFields A parent field to split in case the input field is its child
+    * @return A pair consisting of the parent and the child field parts
+    */
+  private[transformations] def splitByDeepestParent(field: String, parentFields: Seq[String]): (String, String) = {
```

Comment:
Same question:  Why `private`?

Reply:
This function definitely does not cover all use cases that could be provided by a client.